src/core/GLBufferAttribute.js

```diff
@@ -0,0 +1,96 @@
+/**
+ * @author raub / https://github.com/raub
+ */
+
+
+function GLBufferAttribute( gl, buffer, type, itemSize, count, normalized ) {
+
+	this.sizes = [
+		[ gl.FLOAT, 4 ],
+		[ gl.UNSIGNED_SHORT, 2 ],
+		[ gl.SHORT, 2 ],
+		[ gl.UNSIGNED_INT, 4 ],
+		[ gl.INT, 4 ],
+		[ gl.BYTE, 1 ],
+		[ gl.UNSIGNED_BYTE, 1 ],
+	].reduce( function ( accum, current ) {
+
+		accum[ current[ 0 ] ] = current[ 1 ];
+		return accum;
+
+	}, {} );
+
+	if ( ! this.sizes[ type ] ) {
+
+		throw new TypeError( 'THREE.GLBufferAttribute: unsupported GL data type.' );
+
+	}
+
+	this.buffer = buffer;
+	this.type = type;
+	this.itemSize = itemSize;
+	this.elementSize = this.sizes[ type ];
```

Comment:
Is `elementSize` also intended to be private?

Reply:
It is intended to be public. The idea is as follows: you know the type of your data, you pass it, the `elementSize` becomes available, carrying the size-in-bytes value for you.

It is not necessary. We could let the app do all the calculations and leave here only what's necessary for three.js to handle the attribute.

Reply:
Ah wait, I found it here: src\renderers\webgl\WebGLAttributes.js
```
			buffers.set( attribute, {
				buffer: attribute.buffer,
				type: attribute.type,
				bytesPerElement: attribute.elementSize,
				version: attribute.version
			} );
```
so basically it is there for three.js to get the byte size, because there is no `Array` instance to do so (`bytesPerElement: array.BYTES_PER_ELEMENT,`).

Reply:
Um, have you considered to let the user set this value? I mean `GLBufferAttribute` is obviously a low-level API not necessarily intended for beginners. Devs who use it should also know the correct bytes per element.

I would to put as less logic as possible into the ctor and let instead the user defined such information. In this way, you probably don't even need a WebGL rendering context in `GLBufferAttribute` which I find a bit strange anyway.

Reply:
The logic behind this is that passing both `type` and `elementSize` to the constructor is redundant, because the `type` always implies the only possible `elementSize`. So it kind of creates a room for an error to occur when you ask the user to pass both. So I decided to calculate it on the ctor side.

But in general it would be fine either way. If the ctor takes `elementSize`, we get rid of all the logic and the contraversial `this.sizes` member. Also no need for `gl` then. Do you approve of the latter scenario?

Reply:
I would say yes. Keeping `GLBufferAttribute` simple and let the app handle this bit sound like the better design for me. Especially since we can assume the user knows the relationship between `type` and `elementSize` if this class is going to be used.

The final decision is up to @mrdoob though 😉

Reply:
Any updates on that matter? @Mugen87 @mrdoob

Reply:
Let's wait for @mrdoob's feedback.